dog.md

```diff
@@ -0,0 +1,207 @@
+
+module DOG
+    imports KMCD-DRIVER
+    imports PRE-CLIP
+    imports VAT
+    imports VOW
+```
+
+Dog Configuration
+-----------------
+
+```k
+    configuration
+      <dog-state>
+         <dog-wards> .Set      </dog-wards>
+         <dog-ilks>  .Map      </dog-ilks>   // mapping (bytes32 => Ilk) String |-> DogIlk
+         <dog-vat>   0:Address </dog-vat>
+         <dog-vow>   0:Address </dog-vow>
+         <dog-live>  false     </dog-live>
+         <dog-hole>  rad(0)    </dog-hole>
+         <dog-dirt>  rad(0)    </dog-dirt>
+      </dog-state>
+```
+
+```k
+    syntax MCDContract ::= DogContract
+    syntax DogContract ::= "Dog"
+    syntax MCDStep     ::= DogContract "." DogStep [klabel(dogStep)]
+ // ----------------------------------------------------------------
+    rule contract(Dog . _) => Dog
+```
+
+### Constructor
+
+```k
+    syntax DogStep ::= "constructor" Address
+ // ----------------------------------------
+    rule <k> Dog . constructor DOG_VAT => . ... </k>
+         <msg-sender> MSGSENDER </msg-sender>
+         ( <dog-state> _ </dog-state>
+        => <dog-state>
+             <dog-vat>   DOG_VAT            </dog-vat>
+             <dog-live>  true               </dog-live>
+             <dog-wards> SetItem(MSGSENDER) </dog-wards>
+             ...
+           </dog-state>
+         )
+```
+
+Dog Authorization
+-----------------
+
+```k
+    syntax DogStep  ::= DogAuthStep
+    syntax AuthStep ::= DogContract "." DogAuthStep [klabel(dogStep)]
+ // -----------------------------------------------------------------
+    rule [[ wards(Dog) => WARDS ]] <dog-wards> WARDS </dog-wards>
+
+    syntax DogAuthStep ::= WardStep
+ // -------------------------------
+    rule <k> Dog . rely ADDR => . ... </k>
+         <dog-wards> ... (.Set => SetItem(ADDR)) </dog-wards>
+
+    rule <k> Dog . deny ADDR => . ... </k>
+         <dog-wards> WARDS => WARDS -Set SetItem(ADDR) </dog-wards>
+```
+
+Dog Data
+--------
+
+-   `DogIlk` tracks parameters needed for `bite`ing CDPs.
+
+    -   `clip`: Liquidator
+    -   `chop`: Liquidation Penalty
+    -   `hole`: Max DAI needed to cover debt+fees of active auctions per ilk
+    -   `dirt`: Amt DAI needed to cover debt+fees of active auctions per ilk
+
+```k
+    syntax DogIlk ::= Ilk ( clip: Address, chop: Wad, hole: Rad, dirt: Rad ) [klabel(#DogIlk), symbol]
+ // --------------------------------------------------------------------------------------------------
+```
+
+Dog Events
+----------
+
+```k
+    syntax CustomEvent ::= Bark(ilk: String, urn: Address, ink: Wad, art: Wad, due: FInt, clip: Address, Id: Int) [klabel(Bark), symbol]
+ //-------------------------------------------------------------------------------------------------------------------------------------
+
+    syntax DogStep ::= "emitBark" String Address Wad Wad FInt Address Int
+ // --------------------------------------------------------
+    rule <k> BARK_ID:Int ~> emitBark ILK URN INK ART DUE CLIP _ => BARK_ID ... </k>
```

Comment:
@kmbarry1  here on the rule definition of `emitBark` we tell K that we _do not care_ about the concrete value of the integer that comes after `CLIP`, the 0 on the other rules, and in fact the return value is the Integer that is behind the `emitBark`, the `BARK_ID` that returns from the `Clip.kick` call on the 2nd part of the `Bark` rule.

I understand that these circumventions are quite complex if one doesn't read/write K everyday, I used this trick to remain faithful to the solidity implementation of the `Bark `event interface, but I can just simply remove the unused Int from the syntax and semantics so that this doubt does not arise when reading this code, since I bet everyone reading this would be confused. Tell me what you consider best, thanks!

Reply:
Now that I finally (mostyl) understand I'd personally favor removing the unused `Int`...unless that will interfere with refinement proofs in some way (don't think it should though).

Reply:
Could also add a short comment noting that even though the signature doesn't match, the emitted data will be the same.